tests/pre_analysis/test_plate_recording.py

```diff
@@ -287,3 +287,37 @@ def load_se(wf, file_path):
 
     # test full dict
     assert pr.platemap_labels == defaultdict(**new_well_groups)
+
+
+@pytest.mark.parametrize("include_stim_data", [True, False, None])
+def test_PlateRecording_include_stim_data_parameter(mocker, include_stim_data):
+    mocker.patch.object(
+        plate_recording,
+        "find_magnet_positions",
+        autospec=True,
+        side_effect=lambda x, *args, **kwargs: {"X": np.empty((x.shape[-1], 24))},
+    )
+    pr_created_from_h5 = PlateRecording(TEST_VAR_STIM_SESSIONS_FILE_PATH)
+    existing_df = pr_created_from_h5.to_dataframe(include_stim_data=include_stim_data)
+    contains_NA = existing_df.isnull().any().any()
+    if include_stim_data:
+        assert contains_NA
+    else:
+        assert not contains_NA
+
+
+@pytest.mark.parametrize("include_stim_data", [True, False, None])
+def test_PlateRecording_include_stim_data_parameter2(mocker, include_stim_data):
```

Comment:
You can just parametrize the file being used in the test instead of making an entirely new one. The test cases we need are
- a V1 file with stim data
- a V1 file without stim data
- a beta 1 file (will never have stim data)
- an optical file (will never have stim data)